src/Type/Registry/StandardTypeRegistry.php

```diff
@@ -0,0 +1,79 @@
+<?php declare(strict_types=1);
+
+namespace GraphQL\Type\Registry;
+
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Type\Definition\Directive;
+use GraphQL\Type\Definition\ListOfType;
+use GraphQL\Type\Definition\NonNull;
+use GraphQL\Type\Definition\NullableType;
+use GraphQL\Type\Definition\ScalarType;
+use GraphQL\Type\Definition\Type;
+use GraphQL\Type\Introspection;
+
+/**
+ * A registry that returns standard types.
+ */
+interface StandardTypeRegistry
```

Comment:
This interface does more than just deal with standard types. Is it feasible to split:
- `StandardTypeRegistry`
- `TypeWrapper`
- `BuiltInDirectiveRegistry`
- `IntrospectionRegistry`

Or can we find a common term? I think the term type registry is already quite loaded, since we advise users to implement one to provide a `typeLoader` function. I guess most users would not need the whole extent of what this interface does though.

Reply:
Sure, we can split it up. But what will this bring? There are currently only 2 places where the `typeRegistry` is needed: `BuildClientSchema` and `SchemaConfig`. If we split up the interface, we would need to require 4 different types. 

We could remove the `TypeWrapper` functionality (`listOf` and `nonNull`) and keep those on `Type` instead?

Reply:
Removed the TypeWrapper functionality.

Reply:
We could also keep it in a single interface - or perhaps one interface that extends multiple smaller interfaces, but I don't like the name. This interface does many more things than what its name implies.

Reply:
I extracted the directives into BuiltInDirectiveRegistry

But the Introspection one is weird.... 

It's more like a IntrospectionLocator, that points you towards the Introspection that is bound to the same type registry.

Because the Introspection object itself, is a registry for `_schema()` object type + also field definitions (`schemaMetaFieldDef`, `typeMetaFieldDef`, `typeNameMetaFieldDef`).

Reply:
Looks fine now.